src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1709,7 +1709,7 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
 
                 ExecutableCodeBinder bodyBinder = sourceMethod.TryGetBodyBinder();
 
-                if (sourceMethod.IsExtern || sourceMethod.IsDefaultValueTypeConstructor())
+                if (sourceMethod.IsExtern || sourceMethod.IsDefaultValueTypeConstructor(requireZeroInit: false))
```

Comment:
Should this be `requireZeroInit: true` so that we get nullable analysis on initializers (if any)?

Reply:
Any thoughts?

Reply:
It looks like the entire clause is unnecessary. (Instances of `SynthesizedInstanceConstructor` never reach here since we check for `SourceMemberMethodSymbol` above.)

Reply:
Consider keeping an assertion if there are multiple calling paths into this method